### DISCUSSION TITLE
Why is this optional Regex not recognized?

### DISCUSSION THREAD

**Discussion by theo-1** [botman/botman#1408] [Q&A]
This doesn't get recognized and simply goes to my fall-through:

`$botman->hears('/download (([a-zA-Z0-9]{4}\-[a-zA-Z0-9]{4})*)', function...`

It's meant to recognize an (optional) 8 character alphanumeric (with a dash splitting it into 2 segment of 4 characters) after the `download` word.

How else do I achieve this? The following variations also do not work:

`$botman->hears('/download(\s([a-zA-Z0-9]{4}\-[a-zA-Z0-9]{4})*)', function...`

`$botman->hears('/download (\s([a-zA-Z0-9]{4}\-[a-zA-Z0-9]{4})*)', function...`

This is the only thing that works:

`$botman->hears('/download'), function...`

which of course means no parameters.

**Comment by filippotoso** [comment]
This works just fine.

`$botman->hears('common-knowledge-base-trigger(\s+([0-9a-fA-F\-]+))?', function...`

I suggest you check the source code where it's used to verify why your patterns don't work:

https://github.com/botman/botman/blob/2.0/src/Messages/Matcher.php#L45
